src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -267,20 +267,20 @@ private void printDebugGraphTransform(final BaseGraph<?,?> graph, final String f
         }
     }
 
-    private AssemblyResult cleanupSeqGraph(final SeqGraph seqGraph) {
-        printDebugGraphTransform(seqGraph, "sequenceGraph.1.dot");
+    private AssemblyResult cleanupSeqGraph(final SeqGraph seqGraph, final Haplotype refHaplotype) {
+        printDebugGraphTransform(seqGraph, refHaplotype.getLocation() + "-sequenceGraph."+seqGraph.getKmerSize()+".1.0.non_ref_removed.dot");
```

Comment:
Can you refactor so that we don't construct these Strings at all if we're not in debugging mode?

Reply:
I added guarding if statements to the outside of every call to this method.